lib/Subscribers/SubscriberSubscribeController.php

```diff
@@ -215,8 +215,9 @@ private function validateCaptcha($captchaSettings, $data): array {
     return $meta;
   }
 
-  private function getSegmentIds(FormEntity $form, array $data): array {
+  private function getSegmentIds(FormEntity $form, array &$data): array {
```

Comment:
I am not a big fan of passing data using reference. When reading the code that calls this function, it is not obvious that some data are removed here. This function only needs `$data['segments']`, so why are we even passing the whole `$data` array in? Can we rewrite this so that we don't have to pass data using reference?